site/snippets/signers/alchemy/components/EmailSignup.tsx

```diff
@@ -0,0 +1,51 @@
+import { AlchemySigner } from "@alchemy/aa-alchemy";
+import { useMutation } from "@tanstack/react-query";
+import { useMemo } from "react";
+
+export const EmailSignupComponent = () => {
+  // It is recommended you wrap this in React Context or other state management
+  // For this example, we are defining the Alchemy Signer in the component
+  const signer = useMemo(
+    () =>
+      new AlchemySigner({
+        client: {
+          connection: {
+            jwt: "alcht_<KEY>",
+          },
+          iframeConfig: {
+            iframeContainerId: "turnkey-iframe-container",
+          },
+        },
+      }),
+    []
+  );
+
+  // we are using react-query to handle loading states more easily,
+  // feel free to use whatever state management library you prefer
+  const {
+    mutate: signup,
+    isLoading,
+    data: user,
+  } = useMutation({
+    mutationFn: () =>
+      // this will send an email to the user with a link to authenticate
+      signer.authenticate({
+        type: "email",
+        email: "USER_EMAIL",
```

Comment:
let's add the redirect params as part of the example here as well

Reply:
you still need a listener for the bundle for signup and for login. the full flow is:
1. send user email
2. user clicks link in email
3. user hits website which reads the bundle from the url